src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -1897,7 +1897,7 @@ namespace HeatBalanceManager {
 
                 state.dataInputProcessing->inputProcessor->markObjectAsUsed(state.dataHeatBalMgr->CurrentModuleObject, instance.key());
 
-                auto &thisConstruct(state.dataConstruction->Construct(construction_index));
+                auto &thisConstruct = state.dataConstruction->Construct(construction_index);
```

Comment:
If this happened first, could reduce a lot of these locals. e.g.

    thisConstruct.SourceAfterLayer = source_after_layer_number;

to

    thisConstruct.SourceAfterLayer = fields.at("thermal_source_present_after_layer_number").get<int>();